lib/ddtrace/opentracer/tracer.rb

```diff
@@ -65,6 +87,12 @@ def start_active_span(operation_name,
           ignore_active_scope: ignore_active_scope
         )
 
+        # Overwrite the tracer context with the OpenTracing managed context.
+        # This is mostly for the benefit of any out-of-the-box tracing from Datadog,
+        # such that spans generated by that tracing will be attached to the OpenTracer
+        # parent span.
+        datadog_tracer.provider.context = span.datadog_span.context
```

Comment:
This line is responsible for allowing Datadog auto-instrumentation spans to append to traces that were created by OpenTracer. e.g. `open_tracer.span` --> `datadog.span` .